cmake/compilers/MSVC.cmake

```
@@ -21,6 +21,9 @@ string(REGEX REPLACE "/W[0-4]" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
 # Warning suppression C4324: structure was padded due to alignment specifier
 set(TBB_WARNING_LEVEL $<$<NOT:$<CXX_COMPILER_ID:Intel>>:/W4> $<$<BOOL:${TBB_STRICT}>:/WX>)
+if ("${MSVC_CXX_ARCHITECTURE_ID}" STREQUAL "ARM64")
+    set(TBB_ALIGNMENT_WARNING_SUPPRESS /wd4324)
+endif()
 set(TBB_WARNING_SUPPRESS /wd4324)
```

Consider the following approach:
```suggestion
set(TBB_ALIGNMENT_WARNING_SUPPRESS /wd4324)
if ("${MSVC_CXX_ARCHITECTURE_ID}" STREQUAL "ARM64")
    set(TBB_PUBLIC_WARNING_SUPPRESS ${TBB_ALIGNMENT_WARNING_SUPPRESS})
else()
    set(TBB_WARNING_SUPPRESS ${TBB_ALIGNMENT_WARNING_SUPPRESS})
endif()
```